app/views/shared/_nav_auth.html.slim

```diff
@@ -14,7 +14,8 @@ nav.bg-white
       .sm-col-right.sm-right-align
         div = t('shared.nav_auth.welcome', email: current_user.email)
         .mt-12p.h6
-          = link_to t('shared.nav_auth.my_account'), profile_path,
-            class: current_page?(profile_path) ? 'bold gray' : 'underline'
-          span.px1.silver = '|'
+          - if user_fully_authenticated?
```

Comment:
I agree, it would be nicer in the decorator. I started with that idea.

What I found is that the devise controllers have some built-in methods that are part of the definition of `user_fully_authenticated?` and so it was going to require lots of re-writing and messy mixing of controller and model (decorator) to get the business logic right.